xnat_tools/physio_convert.py

```diff
@@ -0,0 +1,301 @@
+    mode_value = np.argwhere(counts == np.max(counts))
+    sample_rate_tics = vals[mode_value].flatten()[0]
+    time_per_tic = 0.0025  # one siemens tic is equal to 2.5ms
+    sample_rate_secs = sample_rate_tics * time_per_tic
+    sample_rate_hz = 1 / sample_rate_secs
+    return sample_rate_tics, sample_rate_hz
+
+
+def incorrect_nan(df, idx):
+    # if there's a value listed as nan, but the values on either side
+    # are not nan and span 2048, it really should have been a 2048 and
+    # the siemens software mistakenly excluded it
+    return np.isnan(df.loc[idx, "DATA"]) and (
+        (df.loc[idx - 1, "DATA"] < 2048 and df.loc[idx + 1, "DATA"] > 2048)
+        or (df.loc[idx - 1, "DATA"] > 2048 and df.loc[idx + 1, "DATA"] < 2048)
+    )
+
+
+def find_matching_bold_filename(bids_func_directory, pmu_filename):
+    keys = pmu_filename.split("_")
+    functional_files = glob.glob(bids_func_directory + "/*_bold.nii*")
+    for k in keys:
+        if k == "func-bold":
+            continue
+        functional_files = [x for x in functional_files if k in x]
+
+    if len(functional_files) > 1:
+        raise Exception("more than one possible matching functional run!")
+    elif len(functional_files) == 0:
+        raise Exception("no matching functional runs!")
+    physio_stub = functional_files[0].split(".")[0].replace("_bold", "")
+
+    return physio_stub
+
+
+@app.command()
+def physio_convert(
+    project: str = typer.Argument(..., help="XNAT's Project ID"),
+    subject: str = typer.Argument(..., help="XNAT's subject ID"),
+    bids_root_dir: str = typer.Argument(..., help="Root output directory for exporting files"),
+    session_suffix: str = typer.Option(
+        "-1",
+        "-S",
+        "--session-suffix",
+        help="The session_suffix is initially set to -1.\
+              This will signify an unspecified session_suffix and default to sess-01.\
+              For multi-session studies, the session label will be pulled from XNAT",
+    ),
+):
+    bids_root_dir = os.path.expanduser(bids_root_dir)
+
+    # Set up working directory
+    if not os.access(bids_root_dir, os.R_OK):
+        raise ValueError("BIDS Root directory must exist")
+
+    # # Paths to export source data in a BIDS friendly way
+    # project, subject, session_suffix = path_string_preprocess(project, subject, session_suffix)
+
+    pi_prefix, study_prefix, subject_prefix, session_prefix = prepare_path_prefixes(
+        project, subject, session_suffix
+    )
+
+    export_dir = f"{bids_root_dir}/{pi_prefix}/{study_prefix}/xnat-export"
+    bids_func_dir = (
+        f"{bids_root_dir}/{pi_prefix}/{study_prefix}/bids/{subject_prefix}/{session_prefix}/func/"
+    )
+    dicom_dir_template = f"{export_dir}/{subject_prefix}/{session_prefix}"
+    pmu_dirs = glob.glob(f"{dicom_dir_template}/*PMU")
+
+    if len(pmu_dirs) == 0:
+        return 0
+
+    else:
+        print(f"Found {len(pmu_dirs)} physio files. Converting to BIDS.")
```

Comment:
```suggestion
        _logger.info(f"Found {len(pmu_dirs)} physio files. Converting to BIDS.")
```
You will need to add `_logger = logging.getLogger(__name__)` at the top of the file, and import logging.

Reply:
Would I also need to take log file and verbosity as parameters and run
`setup_logging(_logger, log_file, verbose_level=verbose)` as you did in bids_postprocess? (Although it looks like there the log file is an empty string and verbosity is always 0). What's the benefit of doing that instead of print?

Reply:
I suppose if we were logging a lot of data from `physio_convert`, it would be helpful to write it to a log file.  In this case, since you're notifying the user there are files to convert (rather than flagging a critical error), it's not necessary to use logging.  Just a good practice, in most cases.  For similar logging statements in `run_heudiconv`, we are using `print`, so I wouldn't worry about changing it.